app/models/payment/braintree/subscription.rb

```diff
@@ -24,4 +24,13 @@ class Payment::Braintree::Subscription < ActiveRecord::Base
   belongs_to :payment_method, class_name: 'Payment::Braintree::PaymentMethod'
   has_many   :transactions,   class_name: 'Payment::Braintree::Transaction',
                               foreign_key: :subscription_id
+
+  def cancel_on_ak(reason)
```

Comment:
@Tuuleh In order to decouple Champaign from the ak-processor, it would be better to frame this method name in terms of "publishing an event" and not "do something on AK". I would think of the "cancellation of a subscription" as a generic event that then ak-processor consumes, but also any other app or component might want to trigger something from this event in the future as well. At this moment, it is only a naming convention, but I think it does make a difference.